R/zchunk_L102.ghg_en_USA_S_T_Y.R

```diff
@@ -31,46 +31,77 @@ module_emissions_L102.ghg_en_USA_S_T_Y_DISABLED <- function(command, ...) {
 
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
-    IEA_flow_sector <- get_data(all_data, "energy/IEA_flow_sector")
-    IEA_product_fuel <- get_data(all_data, "energy/IEA_product_fuel")
+    IEA_flow_sector <- get_data(all_data, "energy/mappings/IEA_flow_sector")
+    IEA_product_fuel <- get_data(all_data, "energy/mappings/IEA_product_fuel")
     EPA_ghg_tech <- get_data(all_data, "emissions/EPA_ghg_tech")
     GCAM_sector_tech <- get_data(all_data, "emissions/GCAM_sector_tech")
-    L101.in_EJ_R_en_Si_F_Yh <- get_data(all_data, "temp-data-inject/L101.in_EJ_R_en_Si_F_Yh")
+    L101.in_EJ_R_en_Si_F_Yh <- get_data(all_data, "L101.in_EJ_R_en_Si_F_Yh")
     EPA_FCCC_GHG_2005 <- get_data(all_data, "emissions/EPA_FCCC_GHG_2005")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Convert EPA GHG emissions inventory to Tg and aggregate by sector and fuel
+    EPA_FCCC_GHG_2005 %>% # start from EPA GHG
+      left_join(EPA_ghg_tech, by = "Source_Category") %>% # define category
+      select(-CO2) %>% # non-CO2 only
+      group_by(sector, fuel) %>%
+      summarize_if(is.numeric, sum, na.rm = FALSE) %>% # sum by sector and fuel
+      filter(!is.na(sector), !is.na(fuel)) %>% # delete NA sectors and fuel
+      mutate_all( funs( replace(., is.na(.), 1))) %>% # placeholder values for existing sectors that has no value.
+      # NOTE: THIS IS A HACK. EPA DOESN'T HAVE EMISSIONS IN SOME SECTORS, WHEN EDGAR DOES. THIS WILL MAKE EMISSIONS PROPORTIONAL TO FUEL USE. NOT SURE IF THIS IS THE BEST STRATEGY.
+      mutate_if(is.numeric, funs(. * CONV_GG_TG)) ->  # Convert to Tg
+      L102.ghg_tg_USA_en_Sepa_F_2005 # GHG balance in 2005
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+    # incorrect fuel name for transport from input mapping.
+    # organize energy balances in USA 2005
+        L101.in_EJ_R_en_Si_F_Yh %>% # start from energy balances
+        filter(GCAM_region_ID == gcam.USA_CODE) %>% # USA region only
+        select(-GCAM_region_ID) %>%
+        gather(variable, energy, -sector, -fuel, -technology) %>%
+        filter(variable == "X2005") %>%  # 2005 data only
```

Comment:
it's for both old data behavior AND new behavior.
i inherited this df from: https://github.com/JGCRI/gcamdata/pull/256
which had Xyear and spreadform.
i could just mutate substr, but I end up throwing away this variable anyway.
how should i proceed?
how should i proceed?